File: solutions/kotlin/01-init/code/src/main/kotlin/Main.kt
import java.io.File

fun main(args: Array<String>) {
   when (args[0]) {
       "init" -> {
           listOf(".git", ".git/objects", ".git/refs").forEach { File(it).mkdir() }
           File(".git/HEAD").writeText("ref: refs/heads/master\n")
           println("Initialized git directory")
       }

       else -> throw IllegalArgumentException("Unknown command: ${args[0]}")
   }
}

File: solutions/typescript/01-init/code/app/main.ts
const args = Deno.args;
const command = args[0];

enum Commands {
    Init = "init",
}

switch (command) {
    case Commands.Init:
        Deno.mkdirSync(".git", { recursive: true });
        Deno.mkdirSync(".git/objects", { recursive: true });
        Deno.mkdirSync(".git/refs", { recursive: true });
        Deno.writeTextFileSync(".git/HEAD", "ref: refs/heads/main\n");
        console.log("Initialized git directory");
        break;
    default:
        throw new Error(`Unknown command ${command}`);
}
